### PR TITLE
Remove or Migrate Usages of Picker

### PR DESCRIPTION
**Pull request by NickGerleman**
The [6/23 RN integration](https://github.com/microsoft/react-native-windows/pull/8357) removes Picker. We still have a few usages of it in custom RNTester examples or the like. This change removes all usages of it from our own code, either removing where inapplicable, or using the reccomended community module instead. This gives us a bit of real coverage of a module in our repo, but we only require it for the RNW RNTester fork instead of including it in RNW itself.

This exposed an issue where we broke usageof RNW with existing community modules (see https://github.com/microsoft/react-native-windows/issues/8430). We workaround this with patch-package for now, which also has us enabling yarn scripts again so we can hook postinstall.

Playground and playground-win32 are both updated to support autolinking, and it is tested that community module picker controls work correctly in them.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8432)

### CODE REVIEW

**Comment by NickGerleman** [comment]
Working on catching us up to upstream RN again atm. Looks like Igor and ACoates are gone, but this blocks us getting up to a fairly old build (6/23). @dannyvv @asklar @jonthysell would appreciate a quick look, since we will need this change before going further with getting back up to date.

**Comment by asklar** [comment]
it seems weird for the RNW repo to depend on what is now a community module. Should we just not be using the picker module and leave the integration tests to run in the picker repo instead? Does RN core rntester tests really use the community module?

**Comment by NickGerleman** [comment]
The usage of Picker in our repo is constrained to some custom RNTester examples. I removed any of our own tessts against Picker in this change to allow for its removal.

Note that RNW itself doesn't depend on the module, just our RNTester fork. I went back and foward on whether to remove our own usages vs use the community module, but adding the usage generated signal of an unintended break, which may be valuable to keep in the future.

**Comment by NickGerleman** [comment]
The usages still here are mostly just for manual test UI. E.g. this Picker in the keyboard focus example page.
image

**Comment by NickGerleman** [comment]
It looks like the Picker community module is not enlightened with regards to WinUI 3, which I guess makes sense...

Will look at patching and PR'ing to add support. I would rather not give up usage of a real community module, just because of the coverage it gives us in terms of finding breaks in our repo towards them.

File: patches/@react-native-picker+picker+1.16.5.patch [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+-      <Private>false</Private>
+-    </ProjectReference>
+-  </ItemGroup>
+   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
++    <ImportGroup Label="ReactNativeWindowsTargets">
++    <Import Project="$(ReactNativeWindowsDir)\PropertySheets\External\Microsoft.ReactNative.Uwp.CppLib.targets" Condition="Exists('$(ReactNativeWindowsDir)\PropertySheets\External\Microsoft.ReactNative.Uwp.CppLib.targets')" />
++  </ImportGroup>
+   <ImportGroup Label="ExtensionTargets">
+     <Import Project="$(SolutionDir)packages\Microsoft.Windows.CppWinRT.2.0.190730.2\build\native\Microsoft.Windows.CppWinRT.targets" Condition="Exists('$(SolutionDir)packages\Microsoft.Windows.CppWinRT.2.0.190730.2\build\native\Microsoft.Windows.CppWinRT.targets')" />
++    <Import Project="$(SolutionDir)\packages\$(WinUIPackageName).$(WinUIPackageVersion)\build\native\$(WinUIPackageName).targets" Condition="Exists('$(SolutionDir)\packages\$(WinUIPackageName).$(WinUIPackageVersion)\build\native\$(WinUIPackageName).targets')" />
```
Comment (NickGerleman) [inline patches/@react-native-picker+picker+1.16.5.patch:89]: Willl PR the change to add WinUI 3 support separately with some cleanup to the Picker repo.

**Review by dannyvv — APPROVED** [review]

**Comment by asklar** [comment]
> The usage of Picker in our repo is constrained to some custom RNTester examples. I removed any of our own tessts against Picker in this change to allow for its removal.
> 
> Note that RNW itself doesn't depend on the module, just our RNTester fork. I went back and foward on whether to remove our own usages vs use the community module, but adding the usage generated signal of an unintended break, which may be valuable to keep in the future.

I think we would still get a signal from the RNgallery app - which we should be updating to new versions anyway. I don't know that having rntester depend on a community repo we don't own is a good idea but I'm happy to be convinced otherwise : )

**Comment by NickGerleman** [comment]
The main limitation of the gallery app is that we only catch issues at branch time, and have a smaller matrix than the main repo.

Wherever possible, I think there is a lot of value in surfacing defects at merge time instead of branch time. It's always a pain wheen we discovered some change broke things a few months ago, and need to randomize folks to go fix it. When we've been talking to core folks, I think that's been the source of a lot of the release delays we've seen.

**Comment by NickGerleman** [comment]
It is possible though that the community module does something problematic to our repo. If that happens I wouldn't be hesitant to remove it as a dependency. In the same style of greater removal we would do now.

**Pull request merged**